src/ballet/shred/fd_shred.h

```diff
@@ -142,6 +142,9 @@ typedef uchar fd_shred_merkle_t[FD_SHRED_MERKLE_NODE_SZ];
 /* Maximum number of data shreds in a slot, also maximum number of parity shreds in a slot */
 #define FD_SHRED_MAX_PER_SLOT (1 << 15UL) /* 32,768 shreds */
 
+/* Maximum number of entry batches per block.  The agave deshredder performs bincode deserialization on a per-batch basis. */
+#define FD_SHRED_MAX_BATCHES_PER_BLOCK (64UL)
+
```

Comment:
> The agave deshredder performs bincode deserialization on a per-batch basis.

From my reading of the code, it first deshreds the individual shreds to reconstruct the original buffer, and then bincode deserializes that buffer _once_ to reconstruct the entries. So we only need to worry about trailing bytes for the single vector of entries, not each batch in the shred.

I could be wrong 😄  where in the code do you see this repeated deserialization?

Reply:
i believe yufeng is saying the same thing as you, which is deserializing a vector of entries = entry batch once

here he's defining that there are up to 64 entry batches in a block. but i believe that constant is wrong. see my comment below

Reply:
Think it's flattened after each individual range gets a deserialize() call?  It's very confusing, but a range corresponds to a batch, and I think deshred() also gets called more than once.

https://github.com/anza-xyz/agave/blob/v2.1.0/ledger/src/blockstore.rs#L3744